src/transformers/models/vit/image_processing_vit.py

```diff
@@ -231,6 +232,9 @@ def preprocess(
         # All transformations expect numpy arrays.
         images = [to_numpy_array(image) for image in images]
 
+        # All transformations expect 3-channel images
```

Comment:
Hmmm...., the difficultly with `to_channel_dimension_format` is the desired behaviour if the input image is grayscale and we do 

```
image = to_channel_dimension_format(image, ChannelDimensionFirst)
```

isn't obvious: should we raise an error? Add a channel? Do nothing? 

However, the reason for this channel setting in `resize` is to convert to a PIL image, which should be possible with grayscale images! I.e. it's the logic in `resize` and `to_pil_image` which should be updated to be compatible with grayscale images.

Reply:
You have a good point. :) 

However, `to_pil_image` is not limited to ViT, and modifying it may affect other models. I am not sure how other models should behave with grayscale images - or even if they should support them.

Now, if the idea is to make only ViT transformations handle single-channel images on their own, we could change the logic  in `resize`, `rescale` and `normalize`. What do you think?